test/mixin/pending-container-test.js

```diff
@@ -0,0 +1,64 @@
+import 'd2l-fetch/d2l-fetch.js';
+import { html, fixture, expect, oneEvent } from '@open-wc/testing';
+import '../utility/siren-sdk-pending-container.js';
+
+window.D2L.Siren.WhitelistBehavior._testMode(true);
+
+describe('PendingContainer', function() {
+
+	// this.timeout(180000);
+
+	let sandbox;
+
+	beforeEach(() => {
+		sandbox = sinon.sandbox.create();
+
+		var organizationEntity = {
+			properties: {
+				name: 'Course Name',
+				code: 'SCI100',
+			},
+			links: [{
+				rel: ['https://api.brightspace.com/rels/parent-semester'],
+				href: '/semester.json'
+			},
+			{
+				rel: ['self'],
+				href: '/organization.json'
+			}]
+		};
+
+		sandbox.stub(window.d2lfetch, 'fetch', (input) => {
+			const whatToFetch = {
+				'/organization.json': organizationEntity
+			};
+			return Promise.resolve({
+				ok: true,
+				json: () => { return Promise.resolve(whatToFetch[input]); }
+			});
+		});
+	});
+
+	afterEach(() => {
+		sandbox.restore();
+	});
+
+	it('waits for children', async() => {
+		const el = await fixture(html`
+				<siren-sdk-pending-container href='/organization.json' token="whatever"></siren-sdk-pending-container>
+			`);
+
+		expect(el).shadowDom.to.equal(`
+				<div>Loading...</div>
+				<siren-sdk-simple-organization hidden="" href="/organization.json" token="whatever"></siren-sdk-simple-organization>
```

Comment:
Is the fact that it's `hidden=""` and not just the boolean attribute `hidden` a quirk of testing here? Seems strange, but based on your description of your experience in working with these tests recently, wouldn't surprise me

Reply:
Yeh, it seems to be a quirk of how lit-element sets these properties. When the boolean attribute is present it seems to get an empty `=""`.

Reply:
When using Lit-Element's `?myattr="${this.someBool}"` directive for boolean attributes, it will either add the attribute as `myattr`, or it won't.  I think what we are seeing here may be the product of the `shadowDom` helper, which may be using `innerHTML` or `outerHTML` properties internally, which will result in `myattr=""`.  For example, inspecting the DOM in the browser console should not show these empty strings.

I think this oddity could be avoided by expecting `el.shadowRoot.querySelector('div').hasAttribute('hidden')` to be `false`, and another expectation statement for the siren-sdk element.